### PR TITLE
Api work

### PR DESCRIPTION
**Pull request by pwesferguson**

### CODE REVIEW

File: imports/api/places.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * @param {number} price 0 <= price <= 4; Google supports a min/max price lvl but
+ * since our current behaviour is to only select one, setting both to the same
+ * @param {string} type https://developers.google.com/places/supported_types
+ */
+export default function getNearbyPlaces(location, radius, price, type) {
+  return googleMapsClient.placesNearby({
+    location,
+    radius,
+    minprice: price,
+    minprice: price,
```
Comment (hailin-zhang) [inline imports/api/places.js]: maxPrice

File: imports/redux/actions/index.jsx [hunk 2]
```diff
@@ -71,16 +83,76 @@ export function addFavourites(favourite) {
...
 
-export function removeFavourites(favouriteToRemove){
+export function removeFavourites(favouriteToRemove) {
   return {
     type: REMOVE_FAVOURITES,
     favouriteToRemove
   };
 }
+
+function requestPlacesStart() {
```
Comment (hailin-zhang) [inline imports/redux/actions/index.jsx]: shouldn't these be exported

File: imports/redux/actions/index.jsx [hunk 3]
```diff
@@ -71,16 +83,76 @@ export function addFavourites(favourite) {
...
+export function getPlaces() {
+  return async (dispatch, getState) => {
+    dispatch(requestPlacesStart());
+    const { location, radius, price, typesAndQuantities, blacklist } = getState();
+    const promises = typesAndQuantities.map(({ type }) => {
+      return getNearbyPlaces(location, radius, price, type);
+    });
+    const listOfPlaces = [];
+    try {
+      promises.forEach(async (promise, i) => {
```
Comment (hailin-zhang) [inline imports/redux/actions/index.jsx]: what is i?
also it gets overwritten in the for loop,  try to name it something more descriptive

File: imports/redux/actions/index.jsx [hunk 4]
```diff
@@ -71,16 +83,76 @@ export function addFavourites(favourite) {
...
+        const response = await promise;
+        if (response.error) throw error;
```
Comment (hailin-zhang) [inline imports/redux/actions/index.jsx]: use {} always

File: imports/redux/actions/index.jsx [hunk 5]
```diff
@@ -71,16 +83,76 @@ export function addFavourites(favourite) {
...
+        const results = response.json.results;
+        const quantity = typesAndQuantities[i].quantity;
+
+        for (let i = 0; i < quantity; i++) {
```
Comment (hailin-zhang) [inline imports/redux/actions/index.jsx]: is it possible to do this without for loops?
also a quick comment describing what this does would be good to have

File: imports/redux/reducers/index.jsx [hunk 6]
```diff
@@ -37,10 +41,19 @@ const initialUserState = {
   userId: ""
 }
 
+const initialPlaceSearchState = {
+  isFetchingPlaces: false,
+  places: [],
+  price: 4,
+  typesAndQuantities: [],
+  error: null
```
Comment (hailin-zhang) [inline imports/redux/reducers/index.jsx]: is there a reason why this is null? undefined is usually the convention

File: imports/redux/actions/index.jsx (hunk 2 above)
Comment (pwesferguson) [reply imports/redux/actions/index.jsx]: I think those action dispatches should always be part of the larger getPlaces action, so afaik it doesn't make sense to export them

File: imports/redux/actions/index.jsx (hunk 5 above)
Comment (pwesferguson) [reply imports/redux/actions/index.jsx]: I didn't see a nice way to do it without for loops, but I'll extract it into a function and clean it up a bit

File: imports/redux/actions/index.jsx (hunk 2 above)
Comment (hailin-zhang) [reply imports/redux/actions/index.jsx]: makes sense!

File: imports/redux/actions/index.jsx (hunk 5 above)
Comment (hailin-zhang) [reply imports/redux/actions/index.jsx]: okay! extracting it is a good idea :)

File: imports/api/places.js (hunk 1 above)
Comment (pwesferguson) [reply imports/api/places.js]: fixed

File: imports/redux/actions/index.jsx (hunk 3 above)
Comment (pwesferguson) [reply imports/redux/actions/index.jsx]: check changes

File: imports/redux/actions/index.jsx (hunk 4 above)
Comment (pwesferguson) [reply imports/redux/actions/index.jsx]: fixed

File: imports/redux/reducers/index.jsx (hunk 6 above)
Comment (pwesferguson) [reply imports/redux/reducers/index.jsx]: fixed

**Pull request merged**